mailbox/event/json/src/test/java/org/apache/james/event/json/MailboxACLUpdatedEventSerializationTest.java

```diff
@@ -0,0 +1,572 @@
+    private static final MailboxACL.Rfc4314Rights RIGHTS = new org.apache.james.mailbox.model.MailboxACL.Rfc4314Rights(MailboxACL.Right.Administer, MailboxACL.Right.Read);
+    private static final EventSerializer EVENT_SERIALIZER = new EventSerializer(new TestId.Factory());
+    private static final String JSON_1 = "{" +
+        "  \"MailboxACLUpdated\":{" +
+        "    \"mailboxPath\":{" +
+        "       \"namespace\":\"#private\"," +
+        "       \"user\":\"bob\"," +
+        "       \"name\":\"mailboxName\"" +
+        "      }," +
+        "    \"aclDiff\":{" +
+        "       \"oldACL\":{\"entries\":{}}," +
+        "       \"newACL\":{\"entries\":{\"$any\":\"ar\", \"alice\":\"i\"}}}," +
+        "    \"mailboxId\":\"23\"," +
+        "    \"sessionId\":6," +
+        "    \"user\":\"user\"" +
+        "   }" +
+        "}";
+
+    private MailboxListener.MailboxACLUpdated mailboxACLUpdatedEvent;
+
+    @BeforeEach
+    void setUp() throws UnsupportedRightException {
+        MailboxACL mailboxACL = MailboxACL.EMPTY.apply(
+            MailboxACL.command()
+                .key(ENTRY_KEY)
+                .rights(RIGHTS)
+                .asAddition())
+            .union(new MailboxACL(new MailboxACL.Entry(MailboxFixture.ALICE, MailboxACL.Right.Insert)));
+        mailboxACLUpdatedEvent = new MailboxListener.MailboxACLUpdated(
+            MailboxSession.SessionId.of(06),
+            USER,
+            new MailboxPath(MailboxConstants.USER_NAMESPACE, "bob", "mailboxName"),
+            ACLDiff.computeDiff(MailboxACL.EMPTY, mailboxACL),
+            TestId.of(23));
+    }
+
+    @Test
+    void mailboxACLUpdatedShouldBeSerialized() {
+        String x = EVENT_SERIALIZER.toJson(mailboxACLUpdatedEvent);
+        System.out.println(x);
+        assertThatJson(EVENT_SERIALIZER.toJson(mailboxACLUpdatedEvent))
+            .isEqualTo(JSON_1);
+    }
+
+    @Test
+    void mailboxACLUpdatedShouldBeDeserialized() {
+        assertThat(EVENT_SERIALIZER.fromJson(JSON_1).get())
+            .isEqualTo(mailboxACLUpdatedEvent);
+    }
+
+    @Test
+    void mailboxACLUpdatedShouldBeDeserializedFromFile() {
+        assertThat(EVENT_SERIALIZER.fromJson(ClassLoaderUtils
+            .getSystemResourceAsString("mailboxACLUpdated.json")).get())
+            .isEqualTo(mailboxACLUpdatedEvent);
+    }
+
+    @Nested
+    class NullUserInMailboxPath {
+        private final String NULL_USER = null;
+        private static final String JSON_2 = "{" +
+            "  \"MailboxACLUpdated\":{" +
+            "    \"mailboxPath\":{" +
+            "       \"namespace\":\"#private\"," +
+            "       \"name\":\"mailboxName\"" +
+            "      }," +
+            "    \"aclDiff\":{" +
+            "       \"oldACL\":{\"entries\":{}}," +
+            "       \"newACL\":{\"entries\":{\"$any\":\"ar\"}}}," +
```

Comment:
I don't like that `entries` field... This level of nesting do not add much.

Can we get rid of it and only have `\"newACL\":{\"$any\":\"ar\"}},` ?

Hint you can do this by changing : 

```
case class ACLDiff(oldACL: MailboxACL, newACL: MailboxACL) {...}
```

to 

```

case class ACLDiff(oldACL: Map[JavaMailboxACL.EntryKey, JavaMailboxACL.Rfc4314Rights], 
    newACL: Map[JavaMailboxACL.EntryKey, JavaMailboxACL.Rfc4314Rights]) {...}
```

(And get rid of the Java MailboxACL POJO at the same time ^^